File: test_commits/python/bug_fix_original5.py
class StatsCalculatorOriginal:
    def __init__(self, numbers):
        self.numbers = numbers

    def get_average(self):
        total = sum(self.numbers)
        return total / len(self.numbers)

    def get_median(self):
        sorted_nums = sorted(self.numbers)
        n = len(sorted_nums)
        mid = n // 2
        if n % 2 == 0:
            return (sorted_nums[mid - 1] + sorted_nums[mid]) / 2
        return sorted_nums[mid]


if __name__ == "__main__":
    data1 = [10, 20, 30, 40]
    stats1 = StatsCalculatorOriginal(data1)
    print(stats1.get_average())
    print(stats1.get_median())

    data2 = []
    stats2 = StatsCalculatorOriginal(data2)
    print(stats2.get_average())
    print(stats2.get_median())

File: test_commits/javascript/bug_fix_original5.js
class StatsCalculator {
    constructor(numbers) {
        this.numbers = numbers;
    }

    getAverage() {
        const sum = this.numbers.reduce((a, b) => a + b);
        return sum / this.numbers.length;
    }

    getMedian() {
        const sorted = [...this.numbers].sort((a, b) => a - b);
        const mid = Math.floor(sorted.length / 2);
        if (sorted.length % 2 === 0) {
            return (sorted[mid - 1] + sorted[mid]) / 2;
        }
        return sorted[mid];
    }
}

const data1 = [10, 20, 30, 40];
const stats1 = new StatsCalculator(data1);
console.log(stats1.getAverage());
console.log(stats1.getMedian());

const data2 = [];
const stats2 = new StatsCalculator(data2);
console.log(stats2.getAverage()); 
console.log(stats2.getMedian());  
